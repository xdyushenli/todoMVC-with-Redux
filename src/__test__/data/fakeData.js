const activeItem = {
    isActive: true,
    id: 1,
    text: 'active item'
}

const completedItem = {
    isActive: false,
    id: 1,
    text: 'completed item'
}

const todoList = [
    {
        isActive: false,
        id: 1,
        text: 'completed item'
    },
    {
        isActive: true,
        id: 2,
        text: 'active item'
    }
]

export {activeItem, completedItem, todoList}