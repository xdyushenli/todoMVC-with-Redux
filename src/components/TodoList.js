import React from 'react';
import TodoItem from './TodoItem';
import '../index.css';

export default class TodoList extends React.Component{
    constructor(props) {
        super(props);
    }

    render() {
        let todoListArr = [...this.props.todoList];
        let showMode = this.props.showMode;

        if(showMode === 'active') {
            todoListArr = todoListArr.filter(item => item.isActive)
        }

        if(showMode === 'completed') {
            todoListArr = todoListArr.filter(item => !item.isActive)
        }

        const todoList = todoListArr.map(((item) => 
            <TodoItem item={item}
                      selectSingle={this.props.selectSingle}
                      changeText={this.props.changeText}
                      deleteItem={this.props.deleteItem}
                      key={item.id}
            ></TodoItem>)
        );

        return (
            <ul id='todo-list'>
                {todoList}
            </ul>
        )
    }
}