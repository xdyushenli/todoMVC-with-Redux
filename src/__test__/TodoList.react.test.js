import React from 'react';
import TodoList from '../components/TodoList';
import renderer from 'react-test-renderer';
import {todoList} from './data/fakeData';

it('render all items', () => {
    const tree = renderer.create
        (
            <TodoList 
                todoList={todoList}
                showMode='all'
            ></TodoList>
        ).toJSON();
    expect(tree).toMatchSnapshot();
})

it('render active items', () => {
    const tree = renderer.create
        (
            <TodoList 
                todoList={todoList}
                showMode='active'
            ></TodoList>
        ).toJSON();
    expect(tree).toMatchSnapshot();
})

it('render completed items', () => {
    const tree = renderer.create
        (
            <TodoList 
                todoList={todoList}
                showMode='completed'
            ></TodoList>
        ).toJSON();
    expect(tree).toMatchSnapshot();
})