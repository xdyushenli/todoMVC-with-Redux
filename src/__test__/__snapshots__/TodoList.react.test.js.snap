// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render active items 1`] = `
<ul
  id="todo-list"
>
  <li
    className="todo-item"
  >
    <button
      className="finish-btn"
      onClick={[Function]}
    />
    <div
      className="item-text"
      onDoubleClick={[Function]}
    >
      active item
    </div>
    <button
      className="delete-btn"
      onClick={[Function]}
    >
      ✕
    </button>
  </li>
</ul>
`;

exports[`render all items 1`] = `
<ul
  id="todo-list"
>
  <li
    className="todo-item"
  >
    <button
      className="finish-btn finished-btn"
      onClick={[Function]}
    />
    <div
      className="item-text finished-text"
      onDoubleClick={[Function]}
    >
      completed item
    </div>
    <button
      className="delete-btn"
      onClick={[Function]}
    >
      ✕
    </button>
  </li>
  <li
    className="todo-item"
  >
    <button
      className="finish-btn"
      onClick={[Function]}
    />
    <div
      className="item-text"
      onDoubleClick={[Function]}
    >
      active item
    </div>
    <button
      className="delete-btn"
      onClick={[Function]}
    >
      ✕
    </button>
  </li>
</ul>
`;

exports[`render completed items 1`] = `
<ul
  id="todo-list"
>
  <li
    className="todo-item"
  >
    <button
      className="finish-btn finished-btn"
      onClick={[Function]}
    />
    <div
      className="item-text finished-text"
      onDoubleClick={[Function]}
    >
      completed item
    </div>
    <button
      className="delete-btn"
      onClick={[Function]}
    >
      ✕
    </button>
  </li>
</ul>
`;
